### ISSUE TITLE
TTA

### ISSUE THREAD

**Issue by a13383622838** [HeinzJiao/PolyR-CNN#10]
Thanks for your codes, I successfully ran it, but I have a question. I saw that TTA is set in config.py, but is TTA actually used in the inference demo.py?

**Comment by HeinzJiao** [comment]
Hi, thanks for your interest in our work! TTA is not used in inference or demo.py.

**Comment by a13383622838** [comment]
ok, thank you. 
1.I have another question: how does the model handle inference with images of different sizes if there is no TTA? I trained on 1400 images of 256×256 for 526379 iterations, and found the results to be very poor when inferring on images of different sizes (217×286, 512×512).
2.What is the role of INPUT:MIN_SIZE_TRAIN in Base-PolyRCNN.yaml?I set it to (256,) during the training process.
3.Where is "models/swin_base_patch4_window7_224_22k.pkl"?

**Comment by HeinzJiao** [comment]
Hi, thanks for your interest!

1. I didn’t test on different image sizes — CrowdAI and Inria use fixed-size images. The model wasn’t designed for scale variation, so performance may drop without resizing.

2. INPUT:MIN_SIZE_TRAIN is unused — it's a leftover from the original Sparse R-CNN config. Sorry for the confusion.

3. The Swin-B pretrained weights can be downloaded here:
https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&ved=2ahUKEwjBgpq7iZuNAxVu_7sIHSAdFs8QFnoECBgQAQ&url=https%3A%2F%2Fgithub.com%2FSwinTransformer%2Fstorage%2Freleases%2Fdownload%2Fv1.0.0%2Fswin_base_patch4_window7_224_22k.pth&usg=AOvVaw2Mr0PHPCaEquYWE0RtPDHr&opi=89978449

By the way, I'm curious — how does the model perform when testing on same-size images in your case? In my recent experiments using the refactored code, I found that when the dataset is much smaller than CrowdAI, the model performs much worse than segmentation-based pipelines.

**Comment by a13383622838** [comment]
Thank you for your answer. I tried to use your model to extract red tide from remote sensing images. When I tested it with 256-sized images, the results were poor, but the performance was good when testing on images that were part of the training set. I have summarized some reasons:
     1-The training set is too small.
     2-Overfitting.
     3-The training set and test set are not in the same region, leading to poor generalization.
     4-This model is not suitable for extracting red tide polygons.
Next, I will extract some test images from the training set to ensure they are from the same region, retrain the model, and then test it again while also trying a different Swin model. Additionally, I would like to ask if there are any polygon extraction models that can perform end-to-end inference on large remote sensing images. I would like to compare those with yours and make modifications.

**Comment by HeinzJiao** [comment]
Thank you for your detailed feedback.

If you're not strictly limited to end-to-end models, I recommend trying segmentation-based methods like HiSup. In my recent experiments, they converge faster, require less data, generalize better, and are more computationally efficient. For cases with more complex and irregular outlines (like red tide), our latest diffusion-based approach LDPoly (https://arxiv.org/abs/2504.20645) may also offer inspiration.

As for end-to-end models, the SOTA approaches I’m familiar with for building outline extraction, PolyBuilding and PolyWorld, are unfortunately not fully accessible: PolyBuilding is not open-sourced, and PolyWorld does not provide training code. But you can check out PolyMapper (https://github.com/lizuoyue/ETH-Thesis) for another early end-to-end polygon extraction method with publicly available code. 
You may find our recent ISPRS paper RoIPoly (https://doi.org/10.1016/j.isprsjprs.2025.03.030) useful. It enables polygon vertex interaction using a transformer architecture. 
I also recommend applying more diverse data augmentations to improve robustness.

Better also check recent TGRS or ISPRS papers from 2024–2025 for newer methods, as I haven’t worked on building-related end-to-end polygon extraction in the past year.

Feel free to reach out if you have more questions, and best of luck with your work!

**Comment by a13383622838** [comment]
> Thank you for your detailed feedback.
> 
> If you're not strictly limited to end-to-end models, I recommend trying segmentation-based methods like HiSup. In my recent experiments, they converge faster, require less data, generalize better, and are more computationally efficient. For cases with more complex and irregular outlines (like red tide), our latest diffusion-based approach LDPoly (https://arxiv.org/abs/2504.20645) may also offer inspiration.
> 
> As for end-to-end models, the SOTA approaches I’m familiar with for building outline extraction, PolyBuilding and PolyWorld, are unfortunately not fully accessible: PolyBuilding is not open-sourced, and PolyWorld does not provide training code. But you can check out PolyMapper (https://github.com/lizuoyue/ETH-Thesis) for another early end-to-end polygon extraction method with publicly available code. You may find our recent ISPRS paper RoIPoly (https://doi.org/10.1016/j.isprsjprs.2025.03.030) useful. It enables polygon vertex interaction using a transformer architecture. I also recommend applying more diverse data augmentations to improve robustness.
> 
> Better also check recent TGRS or ISPRS papers from 2024–2025 for newer methods, as I haven’t worked on building-related end-to-end polygon extraction in the past year.
> 
> Feel free to reach out if you have more questions, and best of luck with your work!

OK！Thank you very very much！

**Comment by a13383622838** [comment]
> Thank you for your detailed feedback.
> 
> If you're not strictly limited to end-to-end models, I recommend trying segmentation-based methods like HiSup. In my recent experiments, they converge faster, require less data, generalize better, and are more computationally efficient. For cases with more complex and irregular outlines (like red tide), our latest diffusion-based approach LDPoly (https://arxiv.org/abs/2504.20645) may also offer inspiration.
> 
> As for end-to-end models, the SOTA approaches I’m familiar with for building outline extraction, PolyBuilding and PolyWorld, are unfortunately not fully accessible: PolyBuilding is not open-sourced, and PolyWorld does not provide training code. But you can check out PolyMapper (https://github.com/lizuoyue/ETH-Thesis) for another early end-to-end polygon extraction method with publicly available code. You may find our recent ISPRS paper RoIPoly (https://doi.org/10.1016/j.isprsjprs.2025.03.030) useful. It enables polygon vertex interaction using a transformer architecture. I also recommend applying more diverse data augmentations to improve robustness.
> 
> Better also check recent TGRS or ISPRS papers from 2024–2025 for newer methods, as I haven’t worked on building-related end-to-end polygon extraction in the past year.
> 
> Feel free to reach out if you have more questions, and best of luck with your work!

Excuse me again, when will your LDPoly paper and code be open sourced, I'm interested in it！

**Comment by HeinzJiao** [comment]
Thank you for your interest. The paper is currently under review, but you may refer to the preprint available on arXiv for details. The code for LDPoly will be released in the coming weeks. I will notify you once it is officially open-sourced.

**Comment by HeinzJiao** [comment]
> > Thank you for your detailed feedback.
> > If you're not strictly limited to end-to-end models, I recommend trying segmentation-based methods like HiSup. In my recent experiments, they converge faster, require less data, generalize better, and are more computationally efficient. For cases with more complex and irregular outlines (like red tide), our latest diffusion-based approach LDPoly (https://arxiv.org/abs/2504.20645) may also offer inspiration.
> > As for end-to-end models, the SOTA approaches I’m familiar with for building outline extraction, PolyBuilding and PolyWorld, are unfortunately not fully accessible: PolyBuilding is not open-sourced, and PolyWorld does not provide training code. But you can check out PolyMapper (https://github.com/lizuoyue/ETH-Thesis) for another early end-to-end polygon extraction method with publicly available code. You may find our recent ISPRS paper RoIPoly (https://doi.org/10.1016/j.isprsjprs.2025.03.030) useful. It enables polygon vertex interaction using a transformer architecture. I also recommend applying more diverse data augmentations to improve robustness.
> > Better also check recent TGRS or ISPRS papers from 2024–2025 for newer methods, as I haven’t worked on building-related end-to-end polygon extraction in the past year.
> > Feel free to reach out if you have more questions, and best of luck with your work!
> 
> Excuse me again, when will your LDPoly paper and code be open sourced, I'm interested in it！

Hi, the LDPoly paper was accepted by ISPRS in October, and the code has already been released, including pretrained models, full training/inference instructions, and a demo. Feel free to use it if it’s helpful. We are also very open to collaboration on vectorized outline extraction, so feel free to reach out via the corresponding author’s email listed in the paper.

**Comment by a13383622838** [comment]
> > > Thank you for your detailed feedback.
> > > If you're not strictly limited to end-to-end models, I recommend trying segmentation-based methods like HiSup. In my recent experiments, they converge faster, require less data, generalize better, and are more computationally efficient. For cases with more complex and irregular outlines (like red tide), our latest diffusion-based approach LDPoly (https://arxiv.org/abs/2504.20645) may also offer inspiration.
> > > As for end-to-end models, the SOTA approaches I’m familiar with for building outline extraction, PolyBuilding and PolyWorld, are unfortunately not fully accessible: PolyBuilding is not open-sourced, and PolyWorld does not provide training code. But you can check out PolyMapper (https://github.com/lizuoyue/ETH-Thesis) for another early end-to-end polygon extraction method with publicly available code. You may find our recent ISPRS paper RoIPoly (https://doi.org/10.1016/j.isprsjprs.2025.03.030) useful. It enables polygon vertex interaction using a transformer architecture. I also recommend applying more diverse data augmentations to improve robustness.
> > > Better also check recent TGRS or ISPRS papers from 2024–2025 for newer methods, as I haven’t worked on building-related end-to-end polygon extraction in the past year.
> > > Feel free to reach out if you have more questions, and best of luck with your work!
> > 
> > 
> > Excuse me again, when will your LDPoly paper and code be open sourced, I'm interested in it！
> 
> Hi, the LDPoly paper was accepted by ISPRS in October, and the code has already been released, including pretrained models, full training/inference instructions, and a demo. Feel free to use it if it’s helpful. We are also very open to collaboration on vectorized outline extraction, so feel free to reach out via the corresponding author’s email listed in the paper.

OK！Thank you very much！
